File: vendor_service/domain/character.py
from domain.class_type import ClassType


class Character:
    def __init__(self, id: str, name: str, class_type: ClassType) -> None:
        self.id = id
        self.class_type = class_type
        self.name = name

    def __dict__(self) -> dict:
        return{
            "characterId": self.id,
            "name": self.name,
            "classType": self.class_type
        }

    def toJson(self) -> dict:
        return self.__dict__()

File: client/lib/src/domain/character.dart
import 'dart:convert';

import 'package:client/src/domain/class_type.dart';
import 'package:client/src/domain/core/serializable.dart';

class Character extends Serializable {
  final String characterId;
  final String name;
  final ClassType classType;

  Character(this.characterId, this.name, this.classType);

  static Character fromMap(Map<String, dynamic> jsonData) {
    return Character(jsonData["characterId"], jsonData['name'],
        ClassType.fromInt(jsonData["classType"]));
  }

  @override
  Map<String, dynamic> toMap() {
    return {
      "characterId": characterId,
      "name": name,
      "classType": classType.value
    };
  }

  @override
  String toJson() {
    return json.encode(toMap());
  }
}
